aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -782,7 +806,8 @@ def __init__(
 
     def _add_cors(self, event: ResponseEventT, cors: CORSConfig):
         """Update headers to include the configured Access-Control headers"""
-        self.response.headers.update(cors.to_dict(event.get_header_value("Origin")))
+        extracted_origin_header = cors.extract_origin_header(event.resolved_headers_field)
```

Comment:
Is there a better place to do this extraction logic?

Ideally this wouldn't be tied to CORS only but in the headers boundary.

PS: on my phone so please forgive me if I'm misreading it

Reply:
I didn't understand this point. Do you suggest I remove `extract_origin_header` from CORS and move this to another class?